src/compiler/moduleNameResolver.ts

```diff
@@ -188,7 +188,10 @@ namespace ts {
 
         let resolvedTypeReferenceDirective: ResolvedTypeReferenceDirective | undefined;
         if (resolved) {
-            resolved = realpath(resolved, host, traceEnabled);
+            if (!options.preserveSymlinks) {
+                resolved = realpath(resolved, host, traceEnabled);

```

Comment:
realpath is declared optional but oyu never check for its functionality. tsc should check if the current host supports realpath, and we should probably be resilient or throw if realpath isn't available.

Also, I think that `realpath` should be camelCase to be consistent with the rest of the APIs.